### DISCUSSION TITLE
mitmproxy as base for SWG

### DISCUSSION THREAD

**Discussion by SinghNanak** [mitmproxy/mitmproxy#7213] [General]
I planning to create Secure Web Gateway which should support TLS/SSL inspection. Is mitmproxy a good starting point. I know there is a lot of scanning, AI/ML requirements for detecting threats, performance bottlenecks and stuff, but for creating a concept it should be helpful. According to mitmproxy documentation:

- Intercept HTTP & HTTPS requests and responses and modify them on the fly
- Blocklist
- Proxy Authentication (for Http)

these feature would be backbone, by adding additional API's and Logic, It can be made to something exciting. I personally have not studied the mitmproxy much and not a expert. That why I am asking you guys for help. Will it be useful? what problems and limitation will we face? Anything I should know.

**Comment by Srijan67** [comment]
Supposing you want to deploy mitmproxy on server side, I would still recommend to go for other mitmproxy service as its architecture is slow and you will feel it all ur traffic will travel through proxy server. 
Also, read this link: [https://discourse.mitmproxy.org/t/handeling-high-load-in-multicore-machine-with-in-line-scripting/51](https://discourse.mitmproxy.org/t/handeling-high-load-in-multicore-machine-with-in-line-scripting/51). Even though its long ago, I still doubt if it can cater thousands of users traffic.
Hope this helps :)

**Reply by SinghNanak** [reply]
@Srijan67 thank you for your response.

**Reply by Srijan67** [reply]
No Problem! 
I would suggest you to tag devs of mitmproxy and ask them if it is sufficient enough to handle thousands of traffics  for thousands of users like squid proxy. Or do they still need to fix the architecture as they have also build local redirect mode which makes all traffic go through proxy server. And what is the use of that if traffic transfer is slow and can't cater thousands of users traffics.

**Reply by mhils** [reply]
> I would suggest you to tag devs of mitmproxy and ask them if it is sufficient enough to handle thousands of traffics for thousands of users like squid proxy. 

Handling hundreds or thousands of users is an explicit non-goal for the project.

**Reply by Srijan67** [reply]
Thanks for the clarification :) 
If possible, can you also look into my query [here](https://github.com/mitmproxy/mitmproxy/discussions/7252)

**Reply by SinghNanak** [reply]
> > I would suggest you to tag devs of mitmproxy and ask them if it is sufficient enough to handle thousands of traffics for thousands of users like squid proxy.
> 
> Handling hundreds or thousands of users is an explicit non-goal for the project.

Even though it isn't the project's objective, I really appreciate Mitmproxy's current condition and capabilities thank you @mhils and other maintainers/contributors.
